src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/info/switches/v2/SwitchValidationResponseV2.java

```diff
@@ -23,14 +25,120 @@
 import lombok.EqualsAndHashCode;
 import lombok.Value;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
 @Value
-@Builder
+@Builder(toBuilder = true)
 @EqualsAndHashCode(callSuper = false)
 @JsonNaming(value = SnakeCaseStrategy.class)
-public class SwitchValidationResponseV2 extends InfoData {
+public class SwitchValidationResponseV2 extends InfoData implements Chunkable<SwitchValidationResponseV2> {
     boolean asExpected;
     GroupsValidationEntryV2 groups;
     MetersValidationEntryV2 meters;
     RulesValidationEntryV2 rules;
     LogicalPortsValidationEntryV2 logicalPorts;
+
+    @Override
+    public List<SwitchValidationResponseV2> split(int chunkSize) {
+        List<SwitchValidationResponseV2> result = new ArrayList<>();
+        SwitchValidationResponseV2Builder current = SwitchValidationResponseV2.builder().asExpected(asExpected);
+        boolean addCurrent = true;
+        int currentSize = chunkSize;
+
+        if (groups != null) {
```

Comment:
I tried to simplify it but I have one problem. We have global variables here like  `current` `addCurrent` `currentSize`
After each addition we need to update it. How we can do it in sub methods? 

How exactly we can use strategy here? have some parent class for all validation entries?

Reply:
>  We have global variables here like current addCurrent currentSize
In theory, we can use inner class like SwitchValidationResponseTemporaryContext and pass, update it and return between methods. But I doubt that this would significantly increase the readability of code.